.github/workflows/meson.yml

```diff
@@ -1,22 +1,57 @@
 name: Meson Build (Linux, macOS)
-# Controls when the action will run.
+
 on:
-  # Triggers the workflow on push or pull request events but only for the development branch
-  push:
-    branches: [ master, development ]
-  release:
+  # Triggers the workflow when manually dispatched
   workflow_dispatch:
+    inputs:
+      upload_artefacts:
+        description: "Upload build artefacts"
+        type: boolean
+        required: false
+        default: false
+      build_type:
+        description: "Build Configuration"
+        type: choice
+        required: false
+        default: "release"
+        options:
+          - "release"
+          - "debug"
+      debug_level:
+        description: "Debug Level"
+        type: choice
+        required: false
+        default: "release"
+        options:
+          - "release"
+          - "minimal"
+          - "full"
+
+  # Triggers the workflow when called by a top-level workflow
+  workflow_call:
+    inputs:
+      upload_artefacts:
+        type: boolean
+        required: false
+        default: false
+      build_type: # "release", "debug"
```

Comment:
I'm curious, why do you have options here listed as a comment, but defined as a category in `workflow_dispatch`?

Reply:
The `inputs` context for the two triggers, `workflow_dispatch` and `workflow_call`, are similar but slightly different. 

The first defines a manual trigger for users to run as they desire and determines the content of the UI element that they use to do so. (shown here)
<img width="1440" alt="image" src="https://github.com/cortex-command-community/Cortex-Command-Community-Project-Source/assets/73866773/81eb9ea2-c4ac-4f90-8fd2-4459cffc6ce5">

In this case, we obviously want a user to be able to select from one of the defined options for build type/debug level, so GitHub provides us with the `choice` input type that uses a finite set of options:
<img width="344" alt="image" src="https://github.com/cortex-command-community/Cortex-Command-Community-Project-Source/assets/73866773/f2452c45-188c-40f2-a99d-f33e4be8f408">

In the case of a workflow_call though, it's invoked by a workflow/API call where there's no way to enforce a strict schema for the inputs. 

Here, for example:
https://github.com/cortex-command-community/Cortex-Command-Community-Project-Source/blob/e3a7c448612fda3de9d2f6f3af2d140b68ef1ccc/.github/workflows/master_build.yaml#L21 

I can place any value in this yaml file, however without an editor extension to detect the workflow it's referencing *and* check which input types it supports that value looks fine until runtime, at which point it's too late.

Unfortunately, there's no way to restrict workflow_call triggers to a discreet set of inputs like we do when calling them manually. 

That comment was there to make sure as a contributor we knew which string values this workflow could be called with, and was pretty much just a remnant.

I'd love your opinion on whether to keep it in some form. There's no way to enforce it as stated, but it may be good info.

Reply:
Okay, this makes total sense since I guess it doesn't spit out an error if you pass garbage to it.
I'm personally fine with the comment, it seems like a fine way to convey information, given you can't enforce limitations.

So yeah, feel free to resolve this and leave it as-is :)